stan/math/rev/scal/fun/inc_beta.hpp

```diff
@@ -22,13 +22,13 @@ class inc_beta_vvv_vari : public op_vvv_vari {
     grad_reg_inc_beta(d_a, d_b, avi_->val_, bvi_->val_, cvi_->val_,
                       digamma(avi_->val_), digamma(bvi_->val_),
                       digamma(avi_->val_ + bvi_->val_),
-                      std::exp(lbeta(avi_->val_, bvi_->val_)));
+                      beta(avi_->val_, bvi_->val_));
 
     avi_->adj_ += adj_ * d_a;
     bvi_->adj_ += adj_ * d_b;
     cvi_->adj_ += adj_ * std::pow(1 - cvi_->val_, bvi_->val_ - 1)
                   * std::pow(cvi_->val_, avi_->val_ - 1)
-                  / std::exp(lbeta(avi_->val_, bvi_->val_));
+                  / beta(avi_->val_, bvi_->val_);
```

Comment:
[optional]
It looks like `beta(avi_->val_, bvi_->val_)` may have already been computed when the `vari` was constructed.  If so, it's worth saving the value as the beta function evaluation is super expensive (three `lgamma` and an `exp`).